skyportal/handlers/api/listing.py

```diff
@@ -0,0 +1,199 @@
+import re
+from marshmallow.exceptions import ValidationError
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    User,
+    Obj,
+    Listing,
+)
+
+
+class ListingHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+            description: get all objects corresponding to a specific user and that match a list name.
+            requestBody:
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      user_id:
+                        type: string
+                      list_name:
+                         type: string
+                         description: |
+                            find all objects saved to this list.
+                            If empty, will return an array of strings
+                            specifying all non-empty lists associated with this user.
```

Comment:
This should be optional, if it is none we should just return the objects from all of the user's lists.

Reply:
Ok but then how do I get all the lists associated with my user? Should be another API end point?